library/camel-kamelets-utils/src/main/java/org/apache/camel/kamelets/utils/format/converter/standard/JsonModelDataType.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.camel.kamelets.utils.format.converter.standard;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.camel.CamelContext;
+import org.apache.camel.CamelContextAware;
+import org.apache.camel.CamelExecutionException;
+import org.apache.camel.Exchange;
+import org.apache.camel.InvalidPayloadException;
+import org.apache.camel.component.jackson.JacksonDataFormat;
+import org.apache.camel.kamelets.utils.format.spi.DataTypeConverter;
+import org.apache.camel.kamelets.utils.format.spi.annotations.DataType;
+import org.apache.camel.util.ObjectHelper;
+
+/**
+ * Data type converter able to unmarshal to given unmarshalType using jackson data format.
+ * <p/>
+ * Unmarshal type should be given as a fully qualified class name in the exchange properties.
+ */
+@DataType(name = "jsonObject", mediaType = "application/json")
+public class JsonModelDataType implements DataTypeConverter, CamelContextAware {
+
+    public static final String DATA_TYPE_MODEL_PROPERTY = "CamelDataTypeModel";
+
+    private CamelContext camelContext;
+
+    private static final ObjectMapper mapper = new ObjectMapper();
+
+    @Override
+    public void convert(Exchange exchange) {
+        if (!exchange.hasProperties() || !exchange.getProperties().containsKey(DATA_TYPE_MODEL_PROPERTY)) {
+            return;
+        }
+
+        ObjectHelper.notNull(camelContext, "camelContext");
+
+        String type = exchange.getProperty(DATA_TYPE_MODEL_PROPERTY, String.class);
+        try (JacksonDataFormat dataFormat = new JacksonDataFormat(mapper, camelContext.getClassResolver().resolveMandatoryClass(type))) {
```

Comment:
This seems quite heavy as you need to create a class from a string for each message
IMHO, as a followup there should be a way to configure the converter properties i.e. in this case I think the type is likely to be a fixed.

Reply:
Also note that the a data format it likely needed to be[ initialized/started ](https://github.com/apache/camel/blob/bf98522eb05c8e477c182f11d15c302e69d0bc0d/components/camel-jackson/src/main/java/org/apache/camel/component/jackson/AbstractJacksonDataFormat.java#L531)

Reply:
@oscerd @lburgazzoli @davsclaus I have simplified the JsonModel data type converter to not use the JackSonDataFormat but a normal ObjectMapper that is shared on a class instance level